cmd/validators.go

```diff
@@ -24,14 +24,17 @@ import (
 	"errors"
 	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 
 	"github.com/Azure/azure-storage-azcopy/common"
 )
 
 func validateFromTo(src, dst string, userSpecifiedFromTo string) (common.FromTo, error) {
-	inferredFromTo := inferFromTo(src, dst)
+	var inferredFromTo common.FromTo
```

Comment:
The goal here is to always respect what the user says, right?  This looks like it would work, but it seems like an odd way to do it, because there are still checks against inferredFromTo below, in the case where userFromTo is not "".  Those checks seem pointless now, since inferredFromTo will always be Unknown when userFromTo has a value.  I'd suggest that all usages of inferredFromTo below line 43 should probably be removed, and the declaration of inferredFromTo should be moved to line 36.

This doesn't affect correctness, but it does affect maintainability, since right now we have one part of the routine working the new way (your changes in this PR) but a whole lot of code and comments further below that appear to work the other way (but will never be triggered).. That's confusing to maintain..